angular_forms/test/validator_directives_test.dart

```diff
@@ -55,6 +55,53 @@ void main() {
       expect(fixture.assertOnlyInstance.staticControl.valid, false);
     });
   });
+
+  group('PatternValidator', () {
+    NgTestFixture<DynamicPatternComponent> fixture;
+
+    setUp(() async {
+      var testBed =
+      NgTestBed.forComponent(ng.DynamicPatternComponentNgFactory);
+      fixture = await testBed.create();
+    });
+
+    tearDown(() => disposeAnyRunningTest());
+
+    Future<void> updateRequired({String pattern}) async {
+      await fixture.update((cmp) => cmp.pattern = pattern);
+      // We have to do this in a separate turn, so that new required value has
+      // propagated.
+      await fixture
+          .update((cmp) => cmp.dynamicControl.control.updateValueAndValidity());
+    }
+
+    bool dynamicControlValid() =>
+        fixture.assertOnlyInstance.dynamicControl.valid;
+
+    test('can be triggered dynamically', () async {
+      expect(dynamicControlValid(), true);
+
+      await updateRequired(pattern: '[A-Za-z]*');
+
+      expect(dynamicControlValid(), true);
+
+      await fixture.update((cmp) => cmp.value = 'abc');
```

Comment:
I would add a case where changing the pattern itself causes the value to go from valid to invalid or vice-versa.